### PR TITLE
Fix SerializedObject.append performance issue

### PR DESCRIPTION
**Pull request by sublimator**
@shekenahglory 

Try with ledger 7501326

### CODE REVIEW

File: src/js/ripple/serializedobject.js [hunk 1]
```diff
@@ -115,9 +119,9 @@ SerializedObject.check_no_missing_fields = function(typedef, obj) {
   if (missing_fields.length > 0) {
     var object_name;
 
-    if (obj.TransactionType !== void(0)) {
+    if (obj.TransactionType !== undefined) {
       object_name = SerializedObject.lookup_type_tx(obj.TransactionType);
-    } else if (obj.LedgerEntryType != null){
+    } else if (obj.LedgerEntryType !== undefined) {
```
Comment (wltsmrz) [inline src/js/ripple/serializedobject.js]: `!= null` catches null and undefined, but I'm guessing this is sufficient

**Comment by wltsmrz** [comment]
LGTM

File: scripts/verify_ledger_json.js [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
   flags.forEach(function(f) {
     // Do we have the flag?
     var flag_index = argv.indexOf('--' + f);
     // normalize the name of the flag
     f = f.replace('-', '_');
-    // opts has Boolean value for normalized flag key
-    opts[f] = !!~flag_index;
-    if (opts[f]) {
+    // opts_ has Boolean value for normalized flag key
+    opts_[f] = !!~flag_index;
```
Comment (clark800) [inline scripts/verify_ledger_json.js]: I know this isn't part of your pull request, but we should try to replace `!!~x` with `x !== -1` or use `_.includes` instead. You don't have to do it now, but I just want to point out these cases.

File: src/js/ripple/serializedobject.js [hunk 3]
```diff
@@ -272,17 +288,17 @@ SerializedObject.prototype.hash = function(prefix) {
...
+  // var Type = stypes[spec[3]];
 
   if (typeof obj[name] !== 'undefined') {
     // ST: Old serialization code
-    //this.append(SerializedObject.get_field_header(Type.id, field_id));
+    // this.append(SerializedObject.get_field_header(Type.id, field_id));
     try {
       // ST: Old serialization code
-      //Type.serialize(this, obj[name]);
+      // Type.serialize(this, obj[name]);
```
Comment (clark800) [inline src/js/ripple/serializedobject.js]: Can we remove all the commented code?

Comment (sublimator) [inline src/js/ripple/serializedobject.js]: k, will lodash it and excise the commented out codes

File: src/js/ripple/serializedobject.js (hunk 1 above)
Comment (sublimator) [inline src/js/ripple/serializedobject.js]: I always use != null, cause I learned js from looking at compiled coffee code. iirc existence operator `?` uses `!= null`. But the linter was crying. Using `if ( bla )` probably maintain semantics better, but it's not a big deal here eh?

**Comment by geertweening** [comment]
@sublimator please squash

**Comment by geertweening** [comment]
LGTM

**Pull request merged**